ledger/eval/eval_test.go

```diff
@@ -600,6 +600,7 @@ type evalTestLedger struct {
 	rewardsPool         basics.Address
 	latestTotals        ledgercore.AccountTotals
 	tracer              logic.EvalTracer
+	boxes               map[string][]byte
```

Comment:
Every test that uses this `evalTestLedger` creeps me out.  It is a mock that maybe has the same behaviour as a real ledger, and maybe not. So every time we extend it to try to do more, it feels even more as though every test should be converted to use a real ledger.  `DoubleLedger` and `TestConsensusRange` have some real benefits in terms of accuracy and testing across changes in old versions.

I'm not going to hold up this PR, but we have to wean ourselves off this thing.

Reply:
Yeah I agree the actual implementation of the test ledger is pretty far from the actual ledger. It does seem difficult to mock since the data retrieved in the actual ledger hits the caches (trackers) first and then disk if the cache misses.

I haven't looked into the `DoubleLedger` or `TestConsensusRange`, but I will do next time I need to test things requiring the ledger.